pkg/driver/dra_hooks.go

```diff
@@ -75,11 +78,42 @@ func (np *NetworkDriver) PublishResources(ctx context.Context) {
 	}
 }
 
+func (np *NetworkDriver) publishResourcesPrometheusMetrics(devices []resourceapi.Device) {
+	rdmaCount := 0
+	for _, device := range devices {
+		if *device.Attributes[apis.AttrRDMA].BoolValue {
+			rdmaCount++
+		}
+	}
+	publishedDevicesTotal.WithLabelValues("rdma").Set(float64(rdmaCount))
+	publishedDevicesTotal.WithLabelValues("non_rdma").Set(float64(len(devices) - rdmaCount))
+}
+
 func (np *NetworkDriver) PrepareResourceClaims(ctx context.Context, claims []*resourceapi.ResourceClaim) (map[types.UID]kubeletplugin.PrepareResult, error) {
 	klog.V(2).Infof("PrepareResourceClaims is called: number of claims: %d", len(claims))
+	start := time.Now()
+	defer func() {
+		draPluginRequestsLatencySeconds.WithLabelValues(methodPrepareResourceClaims).Observe(time.Since(start).Seconds())
+	}()
+	result, err := np.prepareResourceClaims(ctx, claims)
+
+	// identify errors and log metrics
+	isError := false
+	for _, res := range result {
+		if res.Err != nil {
+			isError = true
+			break
+		}
+	}
+	if isError || err != nil {
```

Comment:
Should we check the error on line 98? Generally we don't want to continue logic that relies on the result of an function that returns an error. IMO this is a very different type of error than a resource claim specific error. One is an overall failure to prepare any resource claim and the other is one that is specific to a single one.